src/integration-tests/src/pubsub.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use pubsub::{client::TopicAdmin, model::Topic};
+use rand::{Rng, distr::Alphanumeric};
+
+use crate::Result;
+
+pub async fn basic_topic(builder: pubsub::builder::topic_admin::ClientBuilder) -> Result<()> {
+    // Enable a basic subscriber. Useful to troubleshoot problems and visually
+    // verify tracing is doing something.
+    #[cfg(feature = "log-integration-tests")]
+    let _guard = {
+        use tracing_subscriber::fmt::format::FmtSpan;
+        let subscriber = tracing_subscriber::fmt()
+            .with_level(true)
+            .with_thread_ids(true)
+            .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
+            .finish();
+
+        tracing::subscriber::set_default(subscriber)
+    };
+
+    let project = crate::project_id()?;
+    let client = builder.build().await?;
+    let (topic_admin, topic) = create_test_topic().await?;
```

Comment:
we usually add some kind of cleanup to delete stale topics from previous tests that crashed or otherwise failed to delete any topics they created. Should we do that here?

Reply:
I called that out as something I wanted to do in a followup PR in the description, but can do so in this PR if we want that.

Reply:
The reason for delaying is that (AFAICT) there is not a field that I could use to just check the creation time on a topic so it wasn't quite as straightforward as with the others. And I was planning to add a create time label or something to hack around that

Reply:
Could also attach a schema that has a revision create_time